umbral-pre-python/src/lib.rs

```diff
@@ -15,63 +108,222 @@ impl SecretKey {
             backend: umbral_pre::SecretKey::random(),
         }
     }
+
+    #[staticmethod]
+    pub fn from_bytes(bytes: &[u8]) -> PyResult<Self> {
+        from_bytes(bytes)
+    }
+}
+
+#[pyproto]
+impl PyObjectProtocol for SecretKey {
+    fn __richcmp__(&self, other: PyRef<SecretKey>, op: CompareOp) -> PyResult<bool> {
+        richcmp(self, other, op)
+    }
+
+    fn __bytes__(&self) -> PyResult<PyObject> {
+        to_bytes(self)
+    }
+
+    fn __str__(&self) -> PyResult<String> {
+        Ok(format!("{}:...", Self::name()))
+    }
 }
 
 #[pyclass(module = "umbral")]
-pub struct PublicKey {
-    backend: umbral_pre::PublicKey,
+#[derive(PartialEq)]
+pub struct SecretKeyFactory {
+    backend: umbral_pre::SecretKeyFactory,
+}
+
+impl HasSerializableBackend<umbral_pre::SecretKeyFactory> for SecretKeyFactory {
+    fn as_backend(&self) -> &umbral_pre::SecretKeyFactory {
+        &self.backend
+    }
+
+    fn from_backend(backend: umbral_pre::SecretKeyFactory) -> Self {
+        Self { backend }
+    }
+
+    fn name() -> &'static str {
+        "SecretKeyFactory"
+    }
 }
 
 #[pymethods]
-impl PublicKey {
+impl SecretKeyFactory {
     #[staticmethod]
-    pub fn from_secret_key(sk: &SecretKey) -> Self {
+    pub fn random() -> Self {
         Self {
-            backend: umbral_pre::PublicKey::from_secret_key(&sk.backend),
+            backend: umbral_pre::SecretKeyFactory::random(),
         }
     }
+
+    pub fn secret_key_by_label(&self, label: &[u8]) -> PyResult<SecretKey> {
```

Comment:
Nitpick: Maybe we could have `from_label` here, similar to `PublicKey.from_secret_key`?